barcodescanner/src/main/java/com/mindera/skeletoid/barcodescanner/view/BarcodeScannerFragment.kt

```diff
@@ -0,0 +1,335 @@
+import android.os.Bundle
+import android.os.Handler
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.common.GoogleApiAvailability
+import com.google.android.gms.vision.MultiProcessor
+import com.google.android.gms.vision.barcode.Barcode
+import com.google.android.gms.vision.barcode.BarcodeDetector
+import com.google.android.material.snackbar.Snackbar
+import com.mindera.skeletoid.barcodescanner.R
+import com.mindera.skeletoid.barcodescanner.barcode.BarcodeGraphic
+import com.mindera.skeletoid.barcodescanner.barcode.BarcodeGraphicTracker
+import com.mindera.skeletoid.barcodescanner.barcode.BarcodeTrackerFactory
+import com.mindera.skeletoid.barcodescanner.callback.BarcodeScannerCallback
+import com.mindera.skeletoid.barcodescanner.callback.BarcodeScannerPermissionsCallback
+import com.mindera.skeletoid.barcodescanner.camera.CameraSource
+import com.mindera.skeletoid.barcodescanner.camera.CameraSourcePreview
+import com.mindera.skeletoid.barcodescanner.camera.GraphicOverlay
+import com.mindera.skeletoid.kt.extensions.views.afterDrawn
+import com.mindera.skeletoid.logs.LOG
+import kotlin.math.sqrt
+
+
+@Suppress("Unused")
+class BarcodeScannerFragment(private val barcodeCallback: BarcodeScannerCallback,
+                             private val permissionsCallback: BarcodeScannerPermissionsCallback?) : Fragment(), BarcodeGraphicTracker.BarcodeDetectorListener {
+    companion object {
+        const val BARCODE_FORMATS_BUNDLE_KEY = "barcodeFormats"
+        const val BARCODE_STORAGE_MIN_PERCENTAGE_BUNDLE_KEY = "BARCODE_STORAGE_MIN"
+        const val BARCODE_CAMERA_RESOLUTION_WIDTH_BUNDLE_KEY = "BARCODE_CAMERA_RESOLUTION_WIDTH"
+        const val BARCODE_CAMERA_RESOLUTION_HEIGHT_BUNDLE_KEY = "BARCODE_CAMERA_RESOLUTION_HEIGHT"
+        const val BARCODE_CAMERA_FRAMES_PER_SECOND_BUNDLE_KEY = "BARCODE_CAMERA_FRAMES_PER_SECOND"
+        const val BARCODE_CAMERA_DETECTION_DELAY_BUNDLE_KEY = "BARCODE_CAMERA_DETECTION_DELAY"
+        const val BARCODE_FRAGMENT_TAG = "BARCODESCANNER_FRAGMENT"
+        const val RC_HANDLE_GOOGLE_MOBILE_SERVICES = 9001
+        const val RC_HANDLE_CAMERA_PERMISSION = 2
+    }
+
+    //Flags that represent the barcode formats we want to read.
+    private var barcodeFormats: Int = R.integer.barcodeFormats
+    //Minimum percentage of available storage on the device (check initCamera commentaries)
+    private var deviceStorageMinPercentage = R.integer.deviceStorageMinPercentage
+    //Camera width resolution
+    private var cameraResolutionWidth = R.integer.cameraResolutionWidth
+    //Camera height resolution
+    private var cameraResolutionHeight = R.integer.cameraResolutionHeight
```

Comment:
Pretty sure you cant do this, this will fetch the int reference to the resource.

Reply:
yup, fixed 👍